WordPress/src/main/java/org/wordpress/android/ui/reader/models/ReaderRelatedPostList.java

```diff
@@ -2,17 +2,18 @@
 
 import android.support.annotation.NonNull;
 
-import org.wordpress.android.models.ReaderPost;
-import org.wordpress.android.models.ReaderPostList;
+import org.json.JSONArray;
 
 import java.util.ArrayList;
 
 public class ReaderRelatedPostList extends ArrayList<ReaderRelatedPost> {
 
-    public ReaderRelatedPostList(@NonNull ReaderPostList posts) {
-        for (ReaderPost post: posts) {
-            add(new ReaderRelatedPost(post));
+    public static ReaderRelatedPostList fromJsonPosts(@NonNull JSONArray jsonPosts) {
+        ReaderRelatedPostList posts = new ReaderRelatedPostList();
+        for (int i = 0; i < jsonPosts.length(); i++) {
+            posts.add(ReaderRelatedPost.fromJson(jsonPosts.optJSONObject(i)));
```

Comment:
Didn't notice that on my first pass but, it seems [ReaderRelatedPost.fromJson()](https://github.com/wordpress-mobile/WordPress-Android/pull/4626/files) is not expecting `null` as a parameter but `jsonPosts.optJSONObject` can return one. We prolly need to null check the parameter first, or change the function to accept null, right?